internal/provider/private_endpoint_services_resource.go

```diff
@@ -339,44 +344,55 @@ func (r *privateEndpointServicesResource) ImportState(
 	resource.ImportStatePassthroughID(ctx, path.Root("cluster_id"), req, resp)
 }
 
-func waitForEndpointServicesCreatedFunc(
+func arePrivateEndpointsServicesCreated(
 	ctx context.Context,
-	clusterID string,
+	cluster *client.Cluster,
 	cl client.Service,
-	services *client.PrivateEndpointServices,
-) retry.RetryFunc {
-	return func() *retry.RetryError {
-		traceAPICall("ListPrivateEndpointServices")
-		apiServices, httpResp, err := cl.ListPrivateEndpointServices(ctx, clusterID)
-		if err != nil {
-			if httpResp != nil && httpResp.StatusCode < http.StatusInternalServerError {
-				return retry.NonRetryableError(fmt.Errorf("error getting endpoint services: %s", formatAPIErrorMessage(err)))
-			} else {
-				return retry.RetryableError(fmt.Errorf("encountered a server error while reading endpoint status - trying again: %v", err))
-			}
+) (*client.PrivateEndpointServices, *retry.RetryError) {
+	// Set of all regions.
+	regions := func() map[string]bool {
```

Comment:
nit: I think renaming this could help code readability later in this function
```suggestion
	pendingClusterRegions := func() map[string]bool {
```